Documentation/coding-guidelines/interop-guidelines.md

```diff
@@ -244,3 +249,42 @@ rather than the underlying integral type.
     should be named with the most discoverable name possible; if that name
     is a concept (e.g. Errors), it can be named using managed naming
     guidelines.
+
+
+## UNIX shims
+
+Often, various UNIX flavors offer the same API from the point-of-view of compatibility
+with C/C++ source code, but they do not have the same ABI. e.g. Fields can be laid out
+differently, constants can have different numeric values, etc.
```

Comment:
Another big one is that exports can be named differently, making the DllImport mechanism itself quite challenging.